### ISSUE TITLE
System requirements for CESM

### ISSUE THREAD

**Issue by apcraig** [CESM-Development/cime#20]
The issue is being used to track CIME system requirements

perl v5.10.0 or newer
LibXML
cmake
netcdf 4.3.2 or newer, 4.3.3.1 recommended
pnetcdf 1.6.0 required, 1.6.1 recommended (if using pnetcdf with pio)
python 2.7 or newer
lapack

gmake ?
intel compiler version ??
pgi compiler version ??
MPI version ??

**Comment by jedwards4b** [comment]
netcdf required 4.3.2 or newer, recommended 4.3.3.1
pnetcdf 1.6.0 required 1.6.1 recommended

On Thu, Jun 4, 2015 at 11:01 AM, apcraig notifications@github.com wrote:

> The issue is being used to track CIME system requirements
> 
> Perl v5.10.0
> LibXML
> netcdf version ??
> intel compiler version ??
> pgi compiler version ??
> MPI version ??
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CESM-Development/cime/issues/20.

## 

Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO

**Comment by billsacks** [comment]
pgi: soon will require >= 15.1

**Comment by apcraig** [comment]
is cmake required, when did that come in?  looks like i can't build pio without cmake.  is that correct?

**Comment by jedwards4b** [comment]
cmake is required for pio and unit tests.  That's been a requirement for a
year or more.

On Sat, Jun 6, 2015 at 12:45 PM, apcraig notifications@github.com wrote:

> is cmake required, when did that come in? looks like i can't build pio
> without cmake. is that correct?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CESM-Development/cime/issues/20#issuecomment-109632983
> .

## 

Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO

**Comment by rljacob** [comment]
Just curious but why will pgi >= 15.1 be required?

**Comment by billsacks** [comment]
Because pgi 15.1 is finally approaching something that doesn't cause repeated compiler bugs whenever we change any code.

More specifically: The rework of the CLM-CISM coupling that will soon be coming in to CIME raised yet another compiler error in pre-15 pgi, and it was taking too much time to try to keep finding work-arounds.

**Comment by apcraig** [comment]
Anyone know what the current recommendations are for the pgi compiler?  I have been using pgi 11.8 on many linux clusters up to now.  Will that work with the current version?  If not what are a couple "recommended" versions, prior to the 15.1 requirements upcoming soon?

**Comment by quantheory** [comment]
According to what I wrote on the bug list, 13.9 is the recommendation for CESM 1.2.2 (it just happens to be the version tested; older versions may work):

https://wiki.ucar.edu/display/ccsm/Fortran+Compiler+Bug+List

Past that, it is mostly "newer is better", because we kept bumping versions, and there are few regressions compared to the number of bug fixes. Over the past year and a half, we've filed something like 40-50 bug reports on the PGI compiler, which have been resolved little-by-little. It's hard to point at any one release and say that that one is where everything should work. Though by 15.1, there were only 3 or so open reports, which is why we're thinking of bumping the version to there for supported machines.

I really doubt that any configuration that uses a recent CLM 4.5 or CAM will work with PGI <14. Intel 12.X or 13.X has a pretty good chance of working for most configurations, but we've been sticking to the bleeding edge for PGI. (Plus the PGI reps I've spoken to have actively dissuaded us from using PGI 12 or earlier versions if there is any other option; they are quite aware that there are a lot of issues that have had to be ironed out in the past few years.)
